From 0faad6a2a9ce75681cf32293f50cab8b9142707f Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 7 Oct 2022 10:51:28 +0200
Subject: [PATCH 138/140] arm64: dts: st: Add sysconf for the Cortex M33 boot
 on stm32mp251

To be able to set the Cortex-M33 vector table address, the
stm32-rproc driver needs access to this subsystem.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git eeaf3ad5319f7d7a37fbc3f99a6e4a7f6a478962 arm64: dts: st: Add sysconf for the Cortex M33 boot on stm32mp251]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I6c0b61f8c7a33f75835d4ec7e106832979e862af
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 7b6f714d86113..0747788df5d98 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2329,6 +2329,12 @@ exti2: interrupt-controller@46230000 {
 				<0>,
 				<&intc GIC_SPI 213 IRQ_TYPE_LEVEL_HIGH>;	/* EXTI_70 */
 		};
+
+		a35ss_syscfg: syscon@48802000  {
+			compatible = "st,stm32mp25-a35ss-syscfg", "syscon";
+			reg = <0x48802000 0xac>;
+			status = "disabled";
+		};
 	};
 
 	mlahb: ahb@1 {
-- 
2.51.0

